library/threading.c

```diff
@@ -29,6 +29,14 @@
 
 #include "mbedtls/threading.h"
 
+#if !defined(_WIN32) && (defined(__unix__) || \
+    (defined(__APPLE__) && defined(__MACH__)))
+#include <unistd.h>
```

Comment:
I am not sure whether its necessary to define _POSIX_C_SOURCE before this...

Reply:
Surely we can test this, or check the POSIX standards?

Reply:
[This page](http://pubs.opengroup.org/onlinepubs/007904975/functions/xsh_chap02_02.html) from the POSIX standard says: "A POSIX-conforming application should ensure that the feature test macro _POSIX_C_SOURCE is defined before inclusion of any header." So it looks safer to define it, and the right place would not be here, but at the very top of the file.

As far as testing goes, in practice, with a GCC-Glibc toolchain, this is only necessary when compiling with --std=c99, as otherwise GNU extensions are enabled by default. See also [this comment](https://github.com/ARMmbed/mbedtls/pull/1198/files#diff-6f74e14b83c8d959c7c9f0e898136589R32) in the original PR.

Reply:
Ok, thanks for confirming. I will add the define at the top of this file then.